apps/builder/app/builder/features/settings-panel/props-section/animation/animation-keyframes.tsx

```diff
@@ -208,6 +221,48 @@ export const Keyframes = ({
 
               const newValues = [...keyframes];
               newValues[index] = newValue;
+
```

Comment:
I can't read this code, sorry.

for me, this entire logic needs to be packed into a function and covered with unit tests

as a reader of this component I don't want to read this function, I just want to know high level what it does, not all it's implementation details